testdata/generate_test_data/create_complex_circular_references.py

```diff
@@ -0,0 +1,203 @@
+import os.path
+from typing import Any
+
+from lxml import etree
+
+from dsp_tools import excel2xml
+
+
+def create_and_save_circular_references_test_graph(replication_counter: int = 1, save_location: str = "") -> None:
+    """
+    This function creates a graph with circular references.
+    By default, it saves it in the current directory; this can be changed with the parameter save_location.
+
+    Args:
+        replication_counter: number of times the sub-graphs should be created in one root-graph.
+        save_location: path to the folder, where the file should be saved
+    """
+    root = create_circular_references_test_graph(replication_counter=replication_counter)
+    excel2xml.write_xml(root, os.path.join(save_location, f"test_circular_references_{replication_counter}.xml"))
```

Comment:
```suggestion
    excel2xml.write_xml(root, Path(save_location / f"test_circular_references_{replication_counter}.xml"))
```

It is recommended to use pathlib instead of os.path. Just add `from pathlib import Path`

Reply:
Now it crashes:
TypeError: unsupported operand type(s) for /: 'str' and 'str'
The IDE also tells me: Expected type 'str', got 'Path' instead

Reply:
Sorry, my fault: `Path(save_location) / f"test_circular_references_{replication_counter}.xml"` will work

Reply:
Not with the current linter. Because the save function only allows str. I changed that, now it should be fine.

Reply:
Perfect. Thanks for adapting the type signature of the write_xml function.